manifests/plugin/openscap.pp

```diff
@@ -0,0 +1,45 @@
+# = Foreman OpenSCAP plugin
+#
+# This class installs OpenSCAP plugin
+#
+# === Parameters:
+#
+# $configure_openscap_repo::  Enable custom yum repo with packages needed for foreman_openscap,
+#                             type:boolean
+#
+class foreman::plugin::openscap (
+  $configure_openscap_repo = true,
+) {
+  validate_bool($configure_openscap_repo)
+
+  if $configure_openscap_repo {
+    case $::osfamily {
+      'RedHat': {
+
+        $repo = $::operatingsystem ? {
+          'Fedora' => 'fedora',
+          default  => 'epel',
+        }
+
+        if $::operatingsystemmajrelease == undef {
+          $versions_array = split($::operatingsystemrelease, '\.') # facter 1.6
+          $major = $versions_array[0]
+        } else {
+          $major = $::operatingsystemmajrelease # facter 1.7+
+        }
+
+        yumrepo { 'isimluk-openscap':
+          enabled  => 1,
+          gpgcheck => 0,
+          baseurl  => "http://copr-be.cloud.fedoraproject.org/results/isimluk/OpenSCAP/${repo}-${major}-\$basearch/",
+          before   => [ Foreman::Plugin['openscap'] ],
+        }
+      }
+      default: {
+        fail("Unsupported osfamily ${::osfamily}")
```

Comment:
Sorry to harp on about this, but I think the default value for $configure_openscap_repo should perhaps be determined by a params file so it's only enabled if $::osfamily is 'RedHat'.

At the moment this is going to fail by default on all non-'RedHat' OSes without explicit reconfiguration.

e.g. https://github.com/theforeman/puppet-foreman/blob/master/manifests/params.pp#L33-L38

(sorry, I hadn't realised setting the default to true would result in fail() when I wrote my earlier comments)